File: leetcode/442-FindAllDuplicatesInAnArray.py
class Solution:
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        nums.sort()
        for i in range(len(nums)):
            if i != 0:
                if nums[i] == nums[i-1]:
                    nums[i] = -nums[i]
        nums = list(filter(lambda x: x < 0, nums))
        for i in range(len(nums)):
            nums[i] = -nums[i]
        return nums

File: leetcode/442-FindAllDuplicatesInAnArray.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> res = new ArrayList();
        for(int i = 0;i < nums.length;i++){
            int visitedIndex = Math.abs(nums[i]) - 1;
            if(nums[visitedIndex] < 0){
                res.add(Math.abs(nums[i]));
            }
            else{
                nums[visitedIndex] = -nums[visitedIndex];
            }
        }
        return res;
    }
}
